build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/StartAndroidEmulator.cs

```diff
@@ -136,40 +136,26 @@ void Run (string emulator)
 			p.OutputDataReceived  += output;
 			p.ErrorDataReceived   += error;
 
-			p.Start ();
-			p.BeginOutputReadLine ();
-			p.BeginErrorReadLine ();
-
-			const int Timeout = 20*1000;
-			int i = WaitHandle.WaitAny (new[]{sawError}, millisecondsTimeout: Timeout);
-			if (i == 0 || Log.HasLoggedErrors) {
-				p.Kill ();
-				return;
+			try {
+				p.Start ();
+				EmulatorProcessId = p.Id;
+				p.BeginOutputReadLine ();
+				p.BeginErrorReadLine ();
+
+				const int Timeout = 20*1000;
+				int i = WaitHandle.WaitAny (new[]{sawError}, millisecondsTimeout: Timeout);
+				if (i == 0 || Log.HasLoggedErrors) {
+					p.Kill ();
+					return false;
+				}
+			} finally {
+				p.CancelOutputRead ();
+				p.CancelErrorRead ();
+				p.OutputDataReceived  -= output;
+				p.ErrorDataReceived   -= error;
 			}
-
-			p.CancelOutputRead ();
-			p.CancelErrorRead ();
-
-			p.OutputDataReceived  -= output;
-			p.ErrorDataReceived   -= error;
-
-			p.OutputDataReceived  += WriteProcessOutputMessage;
-			p.ErrorDataReceived   += WriteProcessErrorMessage;
-
-			p.BeginOutputReadLine ();
-			p.BeginErrorReadLine ();
```

Comment:
So, was it specifically these lines that made `Run()` never return on Windows?

Reply:
Yeah this seemed to fix things in CI, though I never had this issue when running locally on Windows strangely enough